File: internal/domains/onboarding/tour.go
package onboarding

import (
	"time"

	"github.com/google/uuid"
)

type TourStatus string

const (
	TourDraft     TourStatus = "draft"
	TourPublished TourStatus = "published"
	TourArchived  TourStatus = "archived"
)

type TriggerType string

const (
	TriggerOnLoad     TriggerType = "on_load"
	TriggerDelay      TriggerType = "delay"
	TriggerExitIntent TriggerType = "exit_intent"
	TriggerManual     TriggerType = "manual"
)

func (t TriggerType) Valid() bool {
	switch t {
	case TriggerOnLoad, TriggerDelay, TriggerExitIntent, TriggerManual:
		return true
	}
	return false
}

type Audience struct {
	ShowOnce bool `json:"show_once"`
	MaxShows int  `json:"max_shows"`
	OnlyNew  bool `json:"only_new"`
}

type Tour struct {
	Id          uuid.UUID   `json:"id" db:"id"`
	Title       string      `json:"title" db:"title"`
	Description string      `json:"description" db:"description"`
	Status      TourStatus  `json:"status" db:"status"`
	TriggerType TriggerType `json:"trigger_type" db:"trigger_type"`
	TargetPath  string      `json:"target_path" db:"target_path"`
	Priority    int         `json:"priority" db:"priority"`
	Audience    Audience    `json:"audience" db:"audience"`
	Hints       []Hint      `json:"hints" db:"hints"`
	CreatedAt   time.Time   `json:"created_at" db:"created_at"`
	UpdatedAt   time.Time   `json:"updated_at" db:"updated_at"`
}

File: frontend/src/types/tour.ts
export type TriggerType = "on_load" | "delay" | "exit_intent" | "manual";

export type Placement = "top" | "bottom" | "left" | "right" | "center";

export interface TriggerConfig {
  delay_ms?: number;
}

export interface Audience {
  show_once: boolean;
  max_shows: number;
  only_new: boolean;
}

export interface TourHint {
  id: string;
  step: number;
  title: string;
  content: string;
  selector?: string;
  placement: Placement;
  page_path?: string;
  spotlight: boolean;
  wait_for_selector: boolean;
  media_url?: string;
}

export interface Tour {
  id: string;
  title: string;
  description: string;
  status?: "draft" | "published";
  target_path: string;
  priority: number;
  trigger_type: TriggerType;
  trigger_config?: TriggerConfig;
  audience: Audience;
  hints: TourHint[];
  current_hint_id?: string;
  version_id?: string;
  updated_at?: string;
  created_at?: string;
}

export interface TourVersion {
  id: string;
  tour_id: string;
  version: number;
  status: "draft" | "published" | "archived";
  trigger_type: TriggerType;
  trigger_config?: TriggerConfig;
  target_path: string;
  audience: Audience;
  created_by?: string;
  created_at: string;
  published_at?: string | null;
  archived_at?: string | null;
  hints?: TourHint[];
}